<a href="https://colab.research.google.com/github/Hamashyach/AnaliseDados_projeto1/blob/main/ProjetoMusicas.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## **Músicas mais tocadas em 2024**

A base de dados Top Music 2024 é um conjunto de dados contemporâneo que reúne informações sobre as músicas mais populares de 2024 em diversas plataformas digitais, como Spotify, YouTube, TikTok, Apple Music, entre outras. Ela contém registros detalhados de métricas de engajamento, como número de execuções, visualizações, curtidas, presença em playlists e índices de popularidade. A base possui 4600 registros e 29 variáveis. Seu objetivo é permitir a análise de padrões que influenciam a popularidade musical e o impacto das redes sociais na difusão das faixas.

**Desafios de análise:**

-Identificar padrões musicais que impulsionam popularidade.

-Avaliar impacto de diferentes mídias sociais na divulgação de músicas.

-Analisar métricas de engajamento ao longo do tempo.

# **Dicionário das variáveis**

**Track Name:** Nome da música.

**Album Name:** Álbum.

**Artist:** Artista(s).

**Release Date:** Data de lançamento.

**ISRC:** Código internacional da faixa.

**All Time Rank:** Posição no ranking histórico.

**Track Score:** Nota atribuída à faixa.

**Spotify Streams:** Execuções no Spotify.

**Spotify Playlist Count:** Presença em playlists do Spotify.

**Spotify Playlist Reach:** Alcance nas playlists do Spotify.

**Spotify Popularity:** Popularidade no Spotify.

**YouTube Views:** Visualizações no YouTube.

**YouTube Likes:** Curtidas no YouTube.

**TikTok Posts:** Publicações no TikTok.

**TikTok Likes:** Curtidas no TikTok.

**TikTok Views:** Visualizações no TikTok.

**YouTube Playlist Reach:** Alcance nas playlists do YouTube.

**Apple Music Playlist Count:** Presença em playlists da Apple Music.

**AirPlay Spins:** Execuções em rádios.

**SiriusXM Spins:** Execuções na SiriusXM.

**Deezer Playlist Count:** Presença em playlists do Deezer.

**Deezer Playlist Reach:** Alcance nas playlists do Deezer.

**Amazon Playlist Count:** Presença em playlists da Amazon.

**Pandora Streams:** Execuções na Pandora.

**Pandora Track Stations:** Estações da Pandora com a faixa.

**Soundcloud Streams:** Execuções no SoundCloud.

**Shazam Counts:** Número de identficações no Shazam.

**TIDAL Popularity:** Popularidade no TIDAL.

**Explicit Track:** Contém conteúdo explícito (Sim/Não).














In [14]:
#pacotes necessários
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import matplotlib.pyplot as plt
from sklearn.metrics import r2_score

In [15]:
sns.set_theme(style="whitegrid", palette='rocket')
# sns.color_palette("flare", as_cmap=True)

In [27]:
#importar o arquivo dos dados
df = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/05_músicas_mais_tocadas.csv', encoding='latin-1')
print("Formato da base (linhas, colunas): ", df.shape)

#Verificar os tipos de dados
print(f'\nOs tipos dos atributos são do tipo:\n{df.dtypes}')

# Visão geral
print("\nColunas disponíveis:")
print(df.columns.tolist())

print("\nPrimeiras linhas da base:")
display(df.head())


Formato da base (linhas, colunas):  (4600, 29)

Os tipos dos atributos são do tipo:
Track                          object
Album Name                     object
Artist                         object
Release Date                   object
ISRC                           object
All Time Rank                  object
Track Score                   float64
Spotify Streams                object
Spotify Playlist Count         object
Spotify Playlist Reach         object
Spotify Popularity            float64
YouTube Views                  object
YouTube Likes                  object
TikTok Posts                   object
TikTok Likes                   object
TikTok Views                   object
YouTube Playlist Reach         object
Apple Music Playlist Count    float64
AirPlay Spins                  object
SiriusXM Spins                 object
Deezer Playlist Count         float64
Deezer Playlist Reach          object
Amazon Playlist Count         float64
Pandora Streams                object
Pand

Track                    Album Name          Artist  \
0         MILLION DOLLAR BABY  Million Dollar Baby - Single   Tommy Richman   
1                 Not Like Us                   Not Like Us  Kendrick Lamar   
2  i like the way you kiss me    I like the way you kiss me         Artemas   
3                     Flowers              Flowers - Single     Miley Cyrus   
4                     Houdini                       Houdini          Eminem   

  Release Date          ISRC All Time Rank  Track Score Spotify Streams  \
0    4/26/2024  QM24S2402528             1        725.4     390,470,936   
1     5/4/2024  USUG12400910             2        545.9     323,703,884   
2    3/19/2024  QZJ842400387             3        538.4     601,309,283   
3    1/12/2023  USSM12209777             4        444.9   2,031,280,633   
4    5/31/2024  USUG12403398             5        423.3     107,034,922   

  Spotify Playlist Count Spotify Playlist Reach  ...  SiriusXM Spins  \
0                 30,716            196,631,588  ...             684   
1                 28,113            174,597,137  ...               3   
2                 54,331            211,607,669  ...             536   
3                269,802            136,569,078  ...           2,182   
4                  7,223            151,469,874  ...               1   

  Deezer Playlist Count Deezer Playlist Reach Amazon Playlist Count  \
0                  62.0            17,598,718                 114.0   
1                  67.0            10,422,430                 111.0   
2                 136.0            36,321,847                 172.0   
3                 264.0            24,684,248                 210.0   
4                  82.0            17,660,624                 105.0   

  Pandora Streams Pandora Track Stations Soundcloud Streams  Shazam Counts  \
0      18,004,655                 22,931          4,818,457      2,669,262   
1       7,780,028                 28,444          6,623,075      1,118,279   
2       5,022,621                  5,639          7,208,651      5,285,340   
3     190,260,277                203,384                NaN     11,822,942   
4       4,493,884                  7,006            207,179        457,017   

  TIDAL Popularity Explicit Track  
0              NaN              0  
1              NaN              1  
2              NaN              0  
3              NaN              0  
4              NaN              1  

[5 rows x 29 columns]

De inicio notamos como o dataset é extenso e com muitas informações.



In [28]:
#Valores nulos ordenados em ordem decrescente
print("\nValores nulos por coluna:")
((df.isnull().sum() / df.shape[0])*100).sort_values(ascending=False)



Valores nulos por coluna:


TIDAL Popularity              100.000000
Soundcloud Streams             72.456522
SiriusXM Spins                 46.152174
Pandora Track Stations         27.565217
TikTok Posts                   25.500000
Pandora Streams                24.043478
Amazon Playlist Count          22.934783
YouTube Playlist Reach         21.934783
TikTok Views                   21.326087
TikTok Likes                   21.304348
Deezer Playlist Reach          20.173913
Deezer Playlist Count          20.021739
Spotify Popularity             17.478261
Shazam Counts                  12.543478
Apple Music Playlist Count     12.195652
AirPlay Spins                  10.826087
YouTube Likes                   6.847826
YouTube Views                   6.695652
Spotify Streams                 2.456522
Spotify Playlist Reach          1.565217
Spotify Playlist Count          1.521739
Artist                          0.108696
All Time Rank                   0.000000
Track Score                     0.000000
ISRC                            0.000000
Track                           0.000000
Album Name                      0.000000
Release Date                    0.000000
Explicit Track                  0.000000
dtype: float64

 Verificamos que existe muitos valores nulos, principalmente na váriável TIDAL Popularity que possui 100% de valores nulos, decidimos então remove-la por completo e fazer uma limpeza desses dados.

In [17]:
#Fazer uma cópia da base
df_limpo = df.copy()

# # Converte as colunas que deveriam ser float (mas sao object por causa da virgula) em verdadeiros floats
colunas_com_virgula = [
    'Spotify Streams', 'Spotify Playlist Reach', 'YouTube Views', 'YouTube Likes',
    'TikTok Posts', 'TikTok Likes', 'TikTok Views', 'YouTube Playlist Reach',
    'Deezer Playlist Reach', 'Pandora Streams', 'Soundcloud Streams', 'Shazam Counts'
]

for col in colunas_com_virgula:
    df_limpo[col] = df_limpo[col].astype(str).str.replace(',', '')
    df_limpo[col] = pd.to_numeric(df_limpo[col], errors='coerce')

# Garante que a data de lancamento esta no tipo data
df_limpo['Release Date'] = pd.to_datetime(df_limpo['Release Date'], errors='coerce')

# Transforma valores binarios para "sim" e "nao"
df_limpo['Explicit Track'] = df_limpo['Explicit Track'].map({0: 'Não', 1: 'Sim'})

# TIDAL Popularity possui 100% de valores nulos, remove-la nao afetara nada
df_limpo.drop(columns=['TIDAL Popularity'], inplace=True)

# Remove linhas com uma quantidade muito grande de nulos
df_limpo = df_limpo[df_limpo.isnull().sum(axis=1) < 10]

# Visualiza a base limpa
print("Base após limpeza: ", df_limpo.shape)
display(df_limpo.head())


Base após limpeza:  (4030, 28)


Track                    Album Name          Artist  \
0         MILLION DOLLAR BABY  Million Dollar Baby - Single   Tommy Richman   
1                 Not Like Us                   Not Like Us  Kendrick Lamar   
2  i like the way you kiss me    I like the way you kiss me         Artemas   
3                     Flowers              Flowers - Single     Miley Cyrus   
4                     Houdini                       Houdini          Eminem   

  Release Date          ISRC All Time Rank  Track Score  Spotify Streams  \
0   2024-04-26  QM24S2402528             1        725.4     3.904709e+08   
1   2024-05-04  USUG12400910             2        545.9     3.237039e+08   
2   2024-03-19  QZJ842400387             3        538.4     6.013093e+08   
3   2023-01-12  USSM12209777             4        444.9     2.031281e+09   
4   2024-05-31  USUG12403398             5        423.3     1.070349e+08   

  Spotify Playlist Count  Spotify Playlist Reach  ...  AirPlay Spins  \
0                 30,716             196631588.0  ...         40,975   
1                 28,113             174597137.0  ...         40,778   
2                 54,331             211607669.0  ...         74,333   
3                269,802             136569078.0  ...      1,474,799   
4                  7,223             151469874.0  ...         12,185   

   SiriusXM Spins  Deezer Playlist Count  Deezer Playlist Reach  \
0             684                   62.0             17598718.0   
1               3                   67.0             10422430.0   
2             536                  136.0             36321847.0   
3           2,182                  264.0             24684248.0   
4               1                   82.0             17660624.0   

   Amazon Playlist Count  Pandora Streams  Pandora Track Stations  \
0                  114.0       18004655.0                  22,931   
1                  111.0        7780028.0                  28,444   
2                  172.0        5022621.0                   5,639   
3                  210.0      190260277.0                 203,384   
4                  105.0        4493884.0                   7,006   

   Soundcloud Streams Shazam Counts Explicit Track  
0           4818457.0     2669262.0            Não  
1           6623075.0     1118279.0            Sim  
2           7208651.0     5285340.0            Não  
3                 NaN    11822942.0            Não  
4            207179.0      457017.0            Sim  

[5 rows x 28 columns]

**Limpeza**


*  Fizemos uma cópia da tabela de dados para não alterar a tabela original

*    Removemos a coluna TIDAL Popularity

* Convertemos valores que eram object(por conta da virgula) em float.

* Garantimos que as datas de lançamento está do tipo data

*  Remove linhas com uma quantidade muito grande de nulos

* Visualizar a base limpa




In [26]:
#Descrição estatísticas dos dados
df.describe()

Track Score  Spotify Popularity  Apple Music Playlist Count  \
count  4600.000000         3796.000000                  4039.00000   
mean     41.844043           63.501581                    54.60312   
std      38.543766           16.186438                    71.61227   
min      19.400000            1.000000                     1.00000   
25%      23.300000           61.000000                    10.00000   
50%      29.900000           67.000000                    28.00000   
75%      44.425000           73.000000                    70.00000   
max     725.400000           96.000000                   859.00000   

       Deezer Playlist Count  Amazon Playlist Count  TIDAL Popularity  \
count            3679.000000            3545.000000               0.0   
mean               32.310954              25.348942               NaN   
std                54.274538              25.989826               NaN   
min                 1.000000               1.000000               NaN   
25%                 5.000000               8.000000               NaN   
50%                15.000000              17.000000               NaN   
75%                37.000000              34.000000               NaN   
max               632.000000             210.000000               NaN   

       Explicit Track  
count     4600.000000  
mean         0.358913  
std          0.479734  
min          0.000000  
25%          0.000000  
50%          0.000000  
75%          1.000000  
max          1.000000

**Análise Estatística Descritiva**

* A popularidade no Spotify é alta, com média de 63 e poucos valores extremos.

* As variáveis de playlists (Apple, Deezer, Amazon) são assimétricas, com poucas músicas em muitas playlists e a maioria com alcance baixo.

* O Track Score mostra que poucas músicas dominam o ranking, com pontuações muito acima da média.

* A variabilidade entre plataformas é grande, indicando que algumas mídias favorecem mais certos artistas.

* A variável TIDAL Popularity está vazia e pode ser descartada.

* Cerca de 36% das músicas são explícitas, segundo a variável Explicit Track.